app/state_machines/configurable_state_machine/machine.rb

```diff
@@ -77,8 +78,56 @@ def event_name(event)
       end
     end
 
+    def get_next_target_state_for_event(event_name, metadata)
+      target_states = get_event_target_states!(event_name)
+      target_states.find do |target_state|
+        call_guards_for_target_state(target_state, metadata)
+      end
+    end
+
+    def call_guards_for_target_state(target_state, metadata)
+      guards = target_state[:guards]
+      guards.blank? ||
+        guards.all? { |g| g.call(object,last_transition,metadata) }
+    end
+
+    #rubocop:disable Metrics/CyclomaticComplexity
+    def next_state_for_event(event, params)
+      user = extract_user_from_metadata(params)
+      if can_trigger_event?(event_name: event, metadata: params)
+        event = event.to_sym
+        role = first_role_that_can_trigger_event_on_case(event_name: event, metadata: params, user: user)
+        user_role_config = @config.user_roles[role]
+        raise InvalidEventError.new(kase: @kase, user: params[:acting_user], event: event, role: role) if user_role_config.nil?
+        state_config = user_role_config.states[@kase.current_state]
+        if state_config.nil? || !state_config.to_hash.keys.include?(event)
+          raise InvalidEventError.new(role: role, kase: @kase, user: params[:acting_user], event: event)
```

Comment:
ditto about adding a `message` to this so that n00bs can better understand the error.